Bottom: 08452cef620327a78c0c1bae781a9e2d9b1a977a
Top:    6d630c408715fcbc0ab440e1f9c058279dbfd9d6
Author: Stanislav Brabec <sbrabec@suse.cz>
Date:   2011-07-18 19:14:19 +0200

Add Toporouter to menu

Toporouter is a very nice and nearly invisible feature. Add it to the
menu to make it more visible.

Affects-bug: lp-812429


---

diff --git a/configure.ac b/configure.ac
index fd14c98..666d51e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -320,6 +320,17 @@ AS_CASE(["x$enable_toporouter"],[xyes | xno],,
 )
 AC_MSG_RESULT([$enable_toporouter])
 AM_CONDITIONAL([WITH_TOPOROUTER], test $enable_toporouter != no)
+AS_CASE([$enable_toporouter],[yes],
+ [
+  topo_enabled=1
+ ],
+ [
+  topo_enabled=0
+ ]
+)
+AC_DEFINE_UNQUOTED([TOPO_ENABLED], [$topo_enabled],
+ [Define to 1 to enable toporouter]
+)
 
 AC_MSG_CHECKING([whether to enable toporouter output])
 AC_ARG_ENABLE([toporouter-output],
diff --git a/src/Makefile.am b/src/Makefile.am
index 7efb2d7..e66ad95 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -230,8 +230,8 @@ EXTRA_DIST= \
 	$(srcdir)/hid/png/hid.conf \
 	$(srcdir)/hid/nelma/hid.conf \
 	$(srcdir)/hid/ps/hid.conf \
-	gpcb-menu.res \
-	pcb-menu.res \
+	gpcb-menu.res.in \
+	pcb-menu.res.in \
 	pcbtest.sh.in \
 	dbus.xml
 
@@ -245,12 +245,26 @@ pcb-menu.h : pcb-menu.res
 	sed 's/\\/\\\\/g; s/"/\\"/g; s/^/"/; s/$$/",/' < ${srcdir}/pcb-menu.res >> $@
 	echo '0};' >> $@
 
+pcb-menu.res : pcb-menu.res.in
+if WITH_TOPOROUTER
+	sed '/%if/d ;/%endif/d' < ${srcdir}/pcb-menu.res.in >> $@
+else
+	sed '/%if/,/%endif/d' < ${srcdir}/pcb-menu.res.in >> $@
+endif
+
 gpcb-menu.h : gpcb-menu.res
 	echo '/* AUTOMATICALLY GENERATED FROM gpcb-menu.res DO NOT EDIT */' > $@
 	echo 'const char *gpcb_menu_default[] = {' >> $@
 	sed 's/\\/\\\\/g; s/"/\\"/g; s/^/"/; s/$$/",/' < ${srcdir}/gpcb-menu.res >> $@
 	echo '0};' >> $@
 
+gpcb-menu.res : gpcb-menu.res.in
+if WITH_TOPOROUTER
+	sed '/%if/d ;/%endif/d' < ${srcdir}/gpcb-menu.res.in >> $@
+else
+	sed '/%if/,/%endif/d' < ${srcdir}/gpcb-menu.res.in >> $@
+endif
+
 hid/common/hidlist.h : Makefile
 	$(MKDIR_P) hid/common
 	true > $@
diff --git a/src/gpcb-menu.res b/src/gpcb-menu.res.in
similarity index 98%
rename from src/gpcb-menu.res
rename to src/gpcb-menu.res.in
index e9469e9..ec39860 100644
--- a/src/gpcb-menu.res
+++ b/src/gpcb-menu.res.in
@@ -326,6 +326,9 @@ MainMenu =
    -
    {"Auto-route selected rats" AutoRoute(Selected)}
    {"Auto-route all rats" AutoRoute(AllRats)}
+%if TOPO_ENABLED
+   {"Toporouter" Toporouter()}
+%endif
    {"Rip up all auto-routed tracks" RipUp(All)}
    -
    {"Optimize routed tracks"
@@ -483,17 +486,17 @@ PopupMenus =
        {"Unselect all objects" Unselect(All)}
        {"Remove selected objects" RemoveSelected()}
        {"Copy selection to buffer"
-	GetXY(Click to set the snap point for this buffer)
-	PasteBuffer(Clear)
-	PasteBuffer(AddSelected)
-	Mode(PasteBuffer)
+        GetXY(Click to set the snap point for this buffer)
+        PasteBuffer(Clear)
+        PasteBuffer(AddSelected)
+        Mode(PasteBuffer)
        }
        {"Cut selection to buffer"
-	GetXY(Click to set the snap point for this buffer)
-	PasteBuffer(Clear)
-	PasteBuffer(AddSelected)
-	RemoveSelected()
-	Mode(PasteBuffer)
+        GetXY(Click to set the snap point for this buffer)
+        PasteBuffer(Clear)
+        PasteBuffer(AddSelected)
+        RemoveSelected()
+        Mode(PasteBuffer)
        }
        {"Convert selection to element" Select(Convert)}
        {"Auto place selected elements" AutoPlaceSelected()}
diff --git a/src/pcb-menu.res b/src/pcb-menu.res.in
similarity index 99%
rename from src/pcb-menu.res
rename to src/pcb-menu.res.in
index aaf48c4..b4189dc 100644
--- a/src/pcb-menu.res
+++ b/src/pcb-menu.res.in
@@ -31,7 +31,7 @@ MainMenu =
    {"About..." About()}
    {"Save layout" Save(Layout) m=S a={"Ctrl-S" "Ctrl<Key>s"}}
    {"Save layout as..." Save(LayoutAs) m=A a={"Shift Ctrl-S" "Shift Ctrl<Key>s"}}
-	{"Revert" Load(Revert,none)}
+   {"Revert" Load(Revert,none)}
    {"Import Schematics" Import() }
    {"Load layout" Load(Layout)}
    {"Load element data to paste-buffer" PasteBuffer(Clear) Load(ElementTobuffer)}
@@ -218,7 +218,7 @@ MainMenu =
     {"Set Dispersion" Import(setdisperse)}
    }
   }
-    
+
   {Select
    {"Select all visible objects" Select(All)}
    {"Select all connected objects" Select(Connection)}
@@ -267,7 +267,7 @@ MainMenu =
    {"Elements" ChangeSquare(SelectedElements)}
    {"Pins" ChangeSquare(SelectedPins)}
   }
-    
+
   {Buffer
    {"Copy selection to buffer" GetXY(Click to set the snap point for this buffer)
     PasteBuffer(Clear) PasteBuffer(AddSelected) Mode(PasteBuffer)}
@@ -295,7 +295,7 @@ MainMenu =
    {"#4" checked=buffer,4 PasteBuffer(4) a={"Shift-4" "Shift<Key>4"}}
    {"#5" checked=buffer,5 PasteBuffer(5) a={"Shift-5" "Shift<Key>5"}}
   }
-    
+
   {Connects
    {"Lookup connection to object" GetXY(Click on the object) Connection(Find) a={"Ctrl-F" "Ctrl<Key>f"}}
    {"Reset scanned pads/pins/vias" Connection(ResetPinsViasAndPads) Display(Redraw)}
@@ -309,6 +309,9 @@ MainMenu =
    -
    {"Auto-route selected rats" AutoRoute(Selected)}
    {"Auto-route all rats" AutoRoute(AllRats)}
+%if TOPO_ENABLED
+   {"Toporouter" Toporouter()}
+%endif
    {"Rip up all auto-routed tracks" RipUp(All)}
    -
    {"Auto-Optimize" djopt(auto)  a={"Shift-=" "Shift<Key>="}}
@@ -331,7 +334,7 @@ MainMenu =
    -
    {"Apply vendor drill mapping" ApplyVendor()}
   }
-    
+
   {Info
    {"Generate object report" ReportObject() a={"Ctrl-R" "Ctrl<Key>r"}}
    {"Generate drill summary" Report(DrillReport)}
